internal/tfengine/tfengine.go

```diff
@@ -150,11 +150,21 @@ func dumpTemplate(conf *Config, pwd, cacheDir, outputPath string, ti *templateIn
 		}
 
 	case ti.ComponentPath != "":
+		// Fetch the component, which could be remote.
 		cp, err := pathutil.Fetch(ti.ComponentPath, pwd, cacheDir)
 		if err != nil {
 			return err
 		}
-		if err := template.WriteDir(cp, outputPath, data); err != nil {
+
+		// If the component path is a single file,
+		// treat the output_path as a file name and only write that out.
+		// Otherwise, treat as directory path
+		write := template.WriteDir
+		if pathutil.IsFile(cp) {
```

Comment:
I am guessing the os.Stat should not raise an error so I would just do it here and return the err if it isn't nil.

Reply:
I was thinking that if it raises an error, that's still a "no". But you're right, it should probably not be swallowed. I'll change it.